lldb-eval/eval.cc

```diff
@@ -250,22 +252,42 @@ void Interpreter::Visit(const ErrorNode*) {
   result_ = Value();
 }
 
-void Interpreter::Visit(const LiteralNode* node) { result_ = node->value(); }
+void Interpreter::Visit(const LiteralNode* node) {
+  struct {
+    Value operator()(llvm::APInt val) {
+      if (type.IsPointerType()) {
+        return CreateValueFromPointer(target, val.getLoBits(64u).getZExtValue(),
```

Comment:
Is it possible for a literal node to be a pointer?

Reply:
Only for null pointer as far as I can see. It looks like that works fine with the path for integers, so I removed the special casing here.